src/main/java/us/tastybento/bskyblock/Settings2.java

```diff
@@ -0,0 +1,1115 @@
+    private int maxTeamSize;
+    private int maxHomes;
+    private int nameMinLength;
+    private int nameMaxLength;
+    private int inviteWait;
+
+    // Reset
+    private int resetLimit;
+    private int resetWait;
+    private boolean leaversLoseReset;
+    private boolean kickedKeepInventory;
+
+    // Remove mobs
+    private boolean removeMobsOnLogin;
+    private boolean removeMobsOnIsland;
+    private List<String> removeMobsWhitelist;
+
+    private boolean makeIslandIfNone;
+    private boolean immediateTeleportOnIsland;
+    private boolean respawnOnIsland;
+
+    // Deaths
+    @ConfigEntry(path = "island.deaths.max")
+    private int deathsMax = 10;
+
+    @ConfigEntry(path = "island.deaths.sum-team")
+    private boolean deathsSumTeam = false;
+
+    // ---------------------------------------------
+
+    /*      PROTECTION      */
+    @ConfigEntry(path = "protection.allow-piston-push")
+    private boolean allowPistonPush = false;
+
+    @ConfigEntry(path = "protection.restrict-flying-mobs")
+    private boolean restrictFlyingMobs = true;
+
+    private int togglePvPCooldown;
+
+    private HashMap<Flag, Boolean> defaultFlags;
+
+    //TODO transform these options below into flags
+    private boolean allowEndermanGriefing;
+    private boolean endermanDeathDrop;
+    private boolean allowTNTDamage;
+    private boolean allowChestDamage;
+    private boolean allowCreeperDamage;
+    private boolean allowCreeperGriefing;
+    private boolean allowMobDamageToItemFrames;
+
+    //TODO flags
+
+    // ---------------------------------------------
+
+    /*      ACID        */
+
+    /*
+     * This settings category only exists if the GameType is ACIDISLAND.
+     */
+
+    @ConfigEntry(path = "acid.damage-op", specificTo = GameType.ACIDISLAND)
+    private boolean acidDamageOp = false;
+
+    @ConfigEntry(path = "acid.damage-chickens", specificTo = GameType.ACIDISLAND)
+    private boolean acidDamageChickens = false;
+
+    @ConfigEntry(path = "acid.options.item-destroy-time", specificTo = GameType.ACIDISLAND)
+    private int acidDestroyItemTime = 0;
+
+    // Damage
+    @ConfigEntry(path = "acid.damage.acid.player", specificTo = GameType.ACIDISLAND)
+    private int acidDamage = 10;
+
+    @ConfigEntry(path = "acid.damage.rain", specificTo = GameType.ACIDISLAND)
+    private int acidRainDamage = 1;
+
+    @ConfigEntry(path = "acid.damage.effects", specificTo = GameType.ACIDISLAND)
+    private List<PotionEffectType> acidEffects = new ArrayList<>(Arrays.asList(PotionEffectType.CONFUSION, PotionEffectType.SLOW));
+
+    /*      SCHEMATICS      */
+    private List<String> companionNames;
+    private ItemStack[] chestItems;
+    private EntityType companionType;
+
+    private boolean useOwnGenerator;
+
+    private HashMap<String,Integer> limitedBlocks;
+    private boolean teamJoinDeathReset;
+    
+    private String uniqueId;
+
+    /**
+     * @return the uniqueId
+     */
+    public String getUniqueId() {
```

Comment:
I know that with the database code, an UUID is needed. But it looks here superfluous. What about an optional UUID ?